allennlp/modules/token_embedders/pretrained_transformer_embedder.py

```diff
@@ -122,6 +125,17 @@ def __init__(
         self._num_added_tokens = self._num_added_start_tokens + self._num_added_end_tokens
 
         if not train_parameters:
+            # Override train in transformer_model to prevent it from changing modes
+            def _train(self, mode):
+                self.training = False
+                for module in self.children():
+                    module.train(False)
+                return self
+
+            setattr(
+                self.transformer_model, "train", types.MethodType(_train, self.transformer_model)
+            )
```

Comment:
Instead of patching the `.train()` method on `self.transformer` model, why not just override the `.train()` method `PretrainedTransformerEmbedder`?

We could implement it like

```python
@overrides
def train(self, mode: bool = True):
    self.training = mode
    for name, module in self.named_children():
        if not self.train_parameters and name == "transformer_model":
            module.eval()
        else:
            module.train(mode)
    return self
```